docs/source/image_process.mdx

```diff
@@ -0,0 +1,127 @@
+# Process image data
+
+🤗 Datasets support loading and processing images with the [`~Image`] feature. This guide will show you how to:
+
+- Load an image dataset.
+- Load a generic image dataset with `ImageFolder`.
+- Add data augmentations to your images.
+- Use [`~Dataset.map`] to quickly apply your transforms to the entire dataset.
+
+## Image datasets
+
+The images in an image dataset are typically either a:
+
+- PIL `image`.
+- Path to an image file that you can load. 
+
+For example, load the [Food-101](https://huggingface.co/datasets/food101) dataset and take a look:
+
+```py
+>>> from datasets import load_dataset, Image
+
+>>> dataset = load_dataset("food101", split="train[100:]")
+>>> dataset[0]
+{'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=384x512 at 0x7FC45AB5C590>,
+ 'label': 6}
+```
+
+The [`~Image`] feature automatically decodes the data from the `image` column to return an image object. Now try and call the `image` column to see what the image is:
+
+```py
+>>> from datasets import load_dataset, Image
+
+>>> dataset = load_dataset("food101", split="train[100:200]")
+>>> dataset[0]["image"]
+```
+
+![image_process_beignet](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/image_process_beignet.png)
+
+
+To load an image from its path, use the [`~Dataset.cast_column`] method. The [`~Image`] feature will decode the data at the path to return an image object:
+
+```py
+>>> from datasets import load_dataset, Image
+
+>>> dataset = load_dataset("cats_vs_dogs").cast_column("image_file_path", Image())
```

Comment:
I find this example a bit weird because here we add a `Image` column to the dataset that already has one. IMO, this would make more sense:
```suggestion
>>> dataset = Dataset.from_dict({"image_file_path": ["path/to/image_1", "path/to/image_2, ..., path/to/image_n"]}).cast_column("image_file_path", Image())
```

Reply:
Good point, I wanted to use a real dataset example but this makes more sense!